### PR TITLE
fixes #20089 - Remove deleted puppet modules

### PR DESCRIPTION
**Pull request by sjha4**
### Steps to reproduce:
1. Sync a puppet URL with modules.
2. Remove a module from the source server.
3. On a fresh sync, the old module that has been removed on the source is still part of the product.

### Changes to fix:
Using remove_missing flag on the pulp end to clone behavior as is on the yum repos.

### CODE REVIEW

**Comment by sjha4** [comment]
[test]

**Comment by sjha4** [comment]
[test]

**Comment by sjha4** [comment]
@jlsherrill : Thanks! Test Cases Passed :+1:  
Removed [WIP] , ready for review.

File: app/models/katello/glue/pulp/repo.rb [hunk 1]
```diff
@@ -157,8 +157,8 @@ def generate_importer(capsule = SmartProxy.default_capsule!)
         when Repository::FILE_TYPE
           Runcible::Models::IsoImporter.new(importer_ssl_options(capsule).merge(:feed => importer_feed_url(capsule)))
         when Repository::PUPPET_TYPE
-          options = {:feed => importer_feed_url(capsule)}
-          Runcible::Models::PuppetImporter.new(importer_ssl_options(capsule).merge(options))
+          #options = {:feed => importer_feed_url(capsule)}
```
Comment (jlsherrill) [inline app/models/katello/glue/pulp/repo.rb]: I think you meant to delete this line?

File: engines/bastion_katello/app/assets/javascripts/bastion_katello/products/details/repositories/new/views/new-repository.html [hunk 2]
```diff
@@ -125,7 +125,7 @@ <h6 translate>
         </p>
       </div>
 
-      <div class="checkbox" ng-show="repository.content_type === 'yum'">
+      <div class="checkbox" ng-show="repository.content_type === 'yum'|| repository.content_type === 'puppet'">
```
Comment (jlsherrill) [inline engines/bastion_katello/app/assets/javascripts/bastion_katello/products/details/repositories/new/views/new-repository.html]: add a space between the || and the repository.content_type  please :)

**Comment by jlsherrill** [comment]
minor issues and nitpicks, but otherwise seems to work fine!

**Comment by sjha4** [comment]
[test]

**Comment by sjha4** [comment]
[test]

File: app/models/katello/glue/pulp/repo.rb (hunk 1 above)
Comment (sjha4) [inline app/models/katello/glue/pulp/repo.rb]: Removed the comment.

File: engines/bastion_katello/app/assets/javascripts/bastion_katello/products/details/repositories/new/views/new-repository.html (hunk 2 above)
Comment (sjha4) [inline engines/bastion_katello/app/assets/javascripts/bastion_katello/products/details/repositories/new/views/new-repository.html]: Modified the spacing.

**Comment by jlsherrill** [comment]
ACK thanks @sjha4 !!

**Pull request merged**